agent/agent.go

```diff
@@ -129,8 +130,12 @@ LOOP:
 				if err := a.runTask(tskCtx, tc); err != nil {
 					return errors.WithStack(err)
 				}
+				needPostGroup = true
```

Comment:
this line doesn't do anything since the check for needPostGroup is in the else block

Reply:
The whole thing is contained in a loop. This sets `needPostGroup` to true so that in the next iteration, if the agent does not receive a task, then the post group commands will run.